spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -458,44 +265,73 @@ public Object doWithRetry(RetryContext context) throws RuntimeException {
 				}
 			}
 		}
-		topicsInUse.put(name, listenedPartitions);
-		ReceivingHandler rh = new ReceivingHandler(properties);
-		rh.setOutputChannel(moduleInputChannel);
+		this.topicsInUse.put(name, listenedPartitions);
+		return listenedPartitions;
+	}
 
-		final FixedSubscriberChannel bridge = new FixedSubscriberChannel(rh);
-		bridge.setBeanName("bridge." + name);
 
+	@Override
+	@SuppressWarnings("unchecked")
+	protected MessageProducer createConsumerEndpoint(String name, String group, Object queue,
```

Comment:
should the "queue" parameter be renamed "destination" (not just here but even more importantly in the superclass?)